.. _home:

Welcome to MAnorm
=================

.. image:: https://travis-ci.org/shao-lab/MAnorm.svg?branch=master
   :alt: Travis Build
   :target: https://travis-ci.org/shao-lab/MAnorm
.. image:: https://readthedocs.org/projects/manorm/badge/?version=latest
   :alt: Documentation Status
   :target: http://manorm.readthedocs.io/en/latest/?badge=latest
.. image:: https://img.shields.io/pypi/v/MAnorm.svg
   :alt: PyPI
   :target: https://pypi.python.org/pypi/MAnorm
.. image:: https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat-square
   :alt: Bioconda
   :target: http://bioconda.github.io/recipes/manorm/README.html
.. image:: https://codecov.io/gh/shao-lab/MAnorm/branch/master/graph/badge.svg
   :alt: Codecov
   :target: https://codecov.io/gh/shao-lab/MAnorm
.. image:: https://img.shields.io/pypi/l/MAnorm.svg
   :alt: License
   :target: https://github.com/shao-lab/MAnorm/blob/master/LICENSE

What is MAnorm?
---------------

**MAnorm** is a robust model for quantitative comparison of ChIP-Seq data sets of TFs (transcription factors)
or epigenetic modifications and you can use it for:

- Normalization of two ChIP-seq samples
- Quantitative comparison (differential analysis) of two ChIP-seq samples
- Evaluating the overlap enrichment of the protein binding sites(peaks)
- Elucidating underlying mechanisms of cell-type specific gene regulation


How MAnorm works?
-----------------

MAnorm uses common peaks of two samples as a reference to build the rescaling model for normalization, which is based
on the empirical assumption that if a chromatin-associated protein has a substantial number of peaks shared in two
conditions, the binding at these common regions will tend to be determined by similar mechanisms, and thus should
exhibit similar global binding intensities across samples. The observed differences on common peaks are presumed to
reflect the scaling relationship of ChIP-Seq signals between two samples, which can be applied to all peaks.

The normalized *M* value given by MAnorm was used as a **quantitative** measure of **differential binding** in each
peak region between two samples, with peak regions associated with larger absolute *M* values exhibiting greater
binding differences between two samples.

.. image:: image/workflow.jpg

MAnorm is developed in Python with a command line interface, you can use it on Linux and macOS platforms.
Basically, it requires a ChIP-seq peak file generated by some peak caller and a sequencing read file for
each sample under comparison. MAnorm supports multiple commonly used input formats, including
BED/MACS/MACS2/narrowPeak formats for peak files and BED/BEDPE/SAM/BAM formats for sequencing read files.

Citation
--------

If you use MAnorm or any derived code, please cite this paper in your publication:

`Shao Z, Zhang Y, Yuan GC, Orkin SH, Waxman DJ. (2012) MAnorm: a robust model for quantitative comparison of ChIP-Seq
data sets. Genome Biol. Mar 16;13(3):R16.`__

.. __: http://genomebiology.com/2012/13/3/R16/abstract

Contents
--------

.. toctree::
   :maxdepth: 1

   Home<self>
   install
   usage
   tutorial
   changelog
   faq
   help
   license

---------------

The Python version of MAnorm is developed by ShaoLab_ at `CAS-MPG Partner Institute for Computational Biology, SIBS,
CAS`_.

.. seealso::
   GitHub repository of MAnorm: https://github.com/shao-lab/MAnorm

.. _ShaoLab: http://bioinfo.sibs.ac.cn/shaolab/
.. _CAS-MPG Partner Institute for Computational Biology, SIBS, CAS: http://www.picb.ac.cn/picb/indexeng.jsp
